osu.Game.Tests/Visual/Gameplay/TestScenePauseWhenInactive.cs

```diff
@@ -33,10 +33,53 @@ public void TestDoesntPauseDuringIntro()
 
             AddStep("resume player", () => Player.GameplayClockContainer.Start());
             AddAssert("ensure not paused", () => !Player.GameplayClockContainer.IsPaused.Value);
+
+            AddStep("progress time to gameplay", () => Player.GameplayClockContainer.Seek(Player.DrawableRuleset.GameplayStartTime));
+            AddUntilStep("wait for pause", () => Player.GameplayClockContainer.IsPaused.Value);
+        }
+
+        /// <summary>
+        /// Tests that if a pause from focus lose is performed while in pause cooldown,
+        /// the player will still pause after the cooldown is finished.
+        /// </summary>
+        [Test]
+        public void TestPauseWhileInCooldown()
+        {
+            AddStep("resume player", () => Player.GameplayClockContainer.Start());
+            AddStep("skip to gameplay", () => Player.GameplayClockContainer.Seek(Player.DrawableRuleset.GameplayStartTime));
+
+            AddStep("set inactive", () => ((Bindable<bool>)host.IsActive).Value = false);
+            AddUntilStep("wait for pause", () => Player.GameplayClockContainer.IsPaused.Value);
+
+            AddStep("set active", () => ((Bindable<bool>)host.IsActive).Value = true);
+
+            AddStep("resume player", () => Player.Resume());
+            AddStep("click resume overlay", () =>
+            {
+                InputManager.MoveMouseTo(this.ChildrenOfType<OsuResumeOverlay.OsuClickToResumeCursor>().Single());
+                InputManager.Click(MouseButton.Left);
+            });
+
+            AddAssert("pause cooldown active", () => Player.PauseCooldownActive);
+            AddStep("set inactive again", () => ((Bindable<bool>)host.IsActive).Value = false);
```

Comment:
I can imagine this failing randomly due to tests running at different timings. You should probably retrieve the `PauseCooldownActive` state in the step where you change the active state, then check it in a following assert.

Reply:
<img width="794" alt="image" src="https://user-images.githubusercontent.com/22781491/108479969-6dab9b80-72a7-11eb-8588-d146a14246b5.png">

like this?

Reply:
yep, that looks better.